### PR TITLE
[For #3649] Fix timeout handling

### PR DESCRIPTION
**Pull request by daipom**
**Which issue(s) this PR fixes**: 
This is about #3609.
Fix timeout handling for #3649.

**What this PR does / why we need it**: 
The current `limit?` usage is complicated, so I have tried to make it simple.

**Docs Changes**:
None.

**Release Note**: 
None.

### CODE REVIEW

File: lib/fluent/plugin_helper/retry_state.rb [hunk 1]
```diff
@@ -44,6 +44,8 @@ def initialize(title, wait, timeout, forever, max_steps, randomize, randomize_wi
 
           @timeout = timeout
           @timeout_at = @start + timeout
+          @has_reached_timeout = false
+          @has_timeouted = false
```
Comment (ashie) [inline lib/fluent/plugin_helper/retry_state.rb:48]:
```suggestion
          @has_timed_out = false
```

File: lib/fluent/plugin_helper/retry_state.rb [hunk 2]
```diff
@@ -123,32 +125,27 @@ def step
...
             @secondary_transition_steps = @steps
           end
+
           @next_time = calc_next_time
+
+          unless @has_reached_timeout
+            @has_reached_timeout = @next_time >= @timeout_at
+          else
+            @has_timeouted = @next_time >= @timeout_at
+          end
```
Comment (ashie) [inline lib/fluent/plugin_helper/retry_state.rb:135]: I don't prefer `unless` clause with `else` clause, double negative makes hard to read:
```suggestion
          if @has_reached_timeout
            @has_timeouted = @next_time >= @timeout_at
          else
            @has_reached_timeout = @next_time >= @timeout_at
          end
```

**Review by ashie — CHANGES_REQUESTED** [review]

**Comment by daipom** [comment]
Thanks for the review. I have reshaped codes, and confirmed the following success again.

- `$ bundle exec rake test TESTOPTS="-t'BufferedOutputRetryTest'"`
- `$ bundle exec rake test TESTOPTS="-t'RetryStateHelperTest'"`

File: lib/fluent/plugin_helper/retry_state.rb (hunk 1 above)
Comment (daipom) [reply lib/fluent/plugin_helper/retry_state.rb]: fixed on 62b7002cdd09b452308e2ef869131f907022d904

File: lib/fluent/plugin_helper/retry_state.rb (hunk 2 above)
Comment (daipom) [reply lib/fluent/plugin_helper/retry_state.rb]: fixed on 62b7002cdd09b452308e2ef869131f907022d904

**Comment by ashie** [comment]
Thanks!

**Pull request merged**